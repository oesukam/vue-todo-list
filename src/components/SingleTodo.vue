<template>
  <div class="todo">
    <div class="todo__index">{{Number(index) + 1}}.</div>
    <div class="todo__text">{{todo.text}}</div>
    <div class="todo__actions">
      <button id="btn-check" class="btn btn-light" @click="checkTodo(todo.id)">
        <i :class="['fas fa-check-circle', {active: todo.done }]"/>
      </button>
      <button id="btn-delete" class="btn btn-light" @click="removeTodo(todo.id)">
        <i class="fa fa-trash"/>
      </button>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    todo: {
      type: Object,
      default: function() {
        return {};
      }
    },
    index: {
      type: Number
    }
  },
  methods: {
    removeTodo(id) {
      this.$emit('removeTodo', id);
    },
    checkTodo(id) {
      this.$emit('checkTodo', id);
    }
  }
};
</script>

<style scoped>
.todo {
  display: flex;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  padding: 10px;
  align-items: center;
  border-radius: 5px;
  margin-bottom: 20px;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
}

.todo__index {
  width: 20px;
  color: #000;
  font-weight: 600;
  white-space: nowrap;
  overflow: hidden;
}

.todo__text {
  flex: 1;
  white-space: nowrap;
  overflow: hidden;
  text-align: left;
  margin-left: 5px;
}

.todo__actions {
  width: 100px;
  display: flex;
  justify-content: flex-end;
}

.todo__actions button i.active {
  color: #209cee;
}

#btn-check {
  margin-right: 10px;
}
</style>
