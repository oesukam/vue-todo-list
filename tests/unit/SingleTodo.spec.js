import { shallowMount } from '@vue/test-utils';
import SingleTodo from '@/components/SingleTodo';

describe('SingleTodo.vue', () => {
  const propsData = {
    todo: {
      id: '6d470672-d137-4333-9ed8-c4939f96b2a5',
      text: 'Testing hints',
      done: false,
      deleting: false
    }
  };

  test('renders correctly', () => {
    const wrapper = shallowMount(SingleTodo, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders correctly without `props`', () => {
    const wrapper = shallowMount(SingleTodo);
    expect(wrapper.props().todo).toEqual({});
  });

  describe('when clicking on `check` todo', () => {
    it('should call `checkTodo`', () => {
      const wrapper = shallowMount(SingleTodo, {
        propsData
      });
      wrapper.find('.btn-check').trigger('click');
      expect(wrapper.props().todo).toHaveProperty('done');
    });
  });

  describe('when clicking on `remove` todo', () => {
    it('should call `removeTodo`', () => {
      const wrapper = shallowMount(SingleTodo, {
        propsData
      });
      wrapper.find('.btn-delete').trigger('click');
      expect(wrapper.props().todo).toHaveProperty('id');
    });
  });
});
