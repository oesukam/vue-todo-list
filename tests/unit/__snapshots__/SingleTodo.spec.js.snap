// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTodo.vue renders correctly 1`] = `
<div
  class="todo"
>
  <div
    class="todo__index"
  >
    NaN.
  </div>
   
  <div
    class="todo__text"
  >
    Testing hints
  </div>
   
  <div
    class="todo__actions"
  >
    <button
      class="btn btn-light"
      id="btn-check"
    >
      <i
        class="fas fa-check-circle"
      />
    </button>
     
    <button
      class="btn btn-light"
      id="btn-delete"
    >
      <i
        class="fa fa-trash"
      />
    </button>
  </div>
</div>
`;
